clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1043,7 +1043,14 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
             /*ExcludeAspects=*/{"fp64"}));
         MPM.addPass(SYCLPropagateJointMatrixUsagePass());
       });
-    else if (LangOpts.SYCLIsHost && !LangOpts.SYCLESIMDBuildHostCode)
+      PB.registerOptimizerEarlyEPCallback([](ModulePassManager &MPM,
+                                             OptimizationLevel Level,
+                                             ThinOrFullLTOPhase) {
+        // Allocate static local memory in SYCL kernel scope for each allocation
+        // call.
```

Comment:
```suggestion
        // Lowers __sycl_allocateLocalMemory  and __sycl_dynamicLocalMemoryPlaceholder
        // builtin calls.
```

Nit, the pass also handles `work_group_scratch_memory` (https://github.com/intel/llvm/pull/15061) but I didn't change this comment as I didn't touch this part.